### PR TITLE
replace prints with logging in rose.server package

### PR DESCRIPTION
**Pull request by moshe742**
see #190

### CODE REVIEW

File: rose/server/game.py [hunk 1]
```diff
@@ -66,7 +69,7 @@ def add_player(self, name):
         self.free_cars.remove(car)
         lane = random.choice(tuple(self.free_lanes))
         self.free_lanes.remove(lane)
-        print 'add player:', name, 'lane:', lane, 'car:', car
+        log.info('add player: %r lane: %r, car: %r', name, lane, car)
```
Comment (nirs) [inline rose/server/game.py]: There is a missing , after the player name.

File: rose/server/game.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import random
 import operator
 from twisted.internet import reactor, task
+import logging
```
Comment (nirs) [inline rose/server/game.py]: Can you move the logging import before the twisted import?

We like to organize imports from less specific to more specific:
1. stdlib imports
1. 3rd party imports
1. rose imports
1. this package imports

File: rose/server/game.py [hunk 3]
```diff
@@ -76,11 +79,12 @@ def remove_player(self, name):
         player = self.players.pop(name)
         self.free_cars.add(player.car)
         self.free_lanes.add(player.lane)
-        print 'remove player:', name, 'lane:', player.lane, 'car:', player.car
+        log.info('remove player: %r, lane: %r, car: %r',
+              name, player.lane, player.car)
```
Comment (nirs) [inline rose/server/game.py]: Can you align the name with the opening parenthesis?

File: rose/server/game.py [hunk 4]
```diff
@@ -76,11 +79,12 @@ def remove_player(self, name):
...
         reactor.callLater(0, self.update_clients)
 
     def drive_player(self, name, info):
-        print 'drive_player:', name, info
+        log.info('drive_player: %r %r', name, info)
```
Comment (nirs) [inline rose/server/game.py]: Looks good

File: rose/server/game.py [hunk 5]
```diff
@@ -92,13 +96,9 @@ def drive_player(self, name, info):
...
     def print_stats(self):
-        print
-        print 'Stats'
+        log.info('Stats')
         for i, p in enumerate(sorted(self.players.values())):
-            print '%d  %10s  row:%d  score:%d' % (
-                i+1, p.name, p.y, p.score
-            )
-        print
+            log.info('%d  %10s  row:%d  score:%d', i + 1, p.name, p.y, p.score)
```
Comment (nirs) [inline rose/server/game.py]: I think we want to format a multiline string and log it, so the message will look like:

```
   game:INFO:Stats
       1   player1  row:0  score:600
       2   player2  row:1  score:590
```

File: rose/server/main.py [hunk 6]
```diff
@@ -8,8 +9,11 @@
 from rose.common import config
 from . import game, net
 
+log = logging.getLogger('main')
```
Comment (nirs) [inline rose/server/main.py]: Please keep 2 blank lines before main to make pep8 tool happy.

File: rose/server/net.py [hunk 7]
```diff
@@ -10,6 +11,9 @@
 from rose.common import error, message
```
Comment (nirs) [inline rose/server/net.py]: We can keep one blank line between the imports and the next line.

File: rose/server/net.py [hunk 8]
```diff
@@ -50,6 +54,7 @@ def broadcast(self, msg):
         for client in self.clients:
             client.send_message(data)
 
+
```
Comment (nirs) [inline rose/server/net.py]: Unrelated change, lets do only logging changes in this patch.

File: rose/server/net.py [hunk 9]
```diff
@@ -66,7 +71,7 @@ def lineReceived(self, line):
             msg = message.parse(line)
             self.dispatch(msg)
         except error.Error as e:
-            print "Error handling message: %s" % e
+            log.info("Error handling message: %s", e)
```
Comment (nirs) [inline rose/server/net.py]: This should  use log.warning() - this is client error.

File: rose/server/net.py [hunk 10]
```diff
@@ -104,6 +109,7 @@ def buildProtocol(self, addr):
         p.factory = self
         return p
 
+
```
Comment (nirs) [inline rose/server/net.py]: Unrelated change.

File: rose/server/score.py [hunk 11]
```diff
@@ -1,7 +1,11 @@
 """ Score logic """
+import logging
 
 from rose.common import actions, config, obstacles
 
+log = logging.getLogger()
```
Comment (nirs) [inline rose/server/score.py]: Needs a logger name "score"

File: rose/server/game.py [hunk 12]
```diff
@@ -96,9 +98,10 @@ def drive_player(self, name, info):
         self.players[name].response_time = info.get('response_time', 1.0)
 
     def print_stats(self):
-        log.info('Stats')
+        string = 'Stats' + os.linesep
         for i, p in enumerate(sorted(self.players.values())):
-            log.info('%d  %10s  row:%d  score:%d', i + 1, p.name, p.y, p.score)
+            string += '%d  %10s  row:%d  score:%d'.format(i + 1, p.name, p.y, p.score)
```
Comment (nirs) [inline rose/server/game.py]: I think we better keep the current string formatting. The good old % placeholder works in every language, and I don't have time to look up in python docs how use the "better" format of the year :-)

Also, we are missing a newline after the first line, but note that we do not want a new line after the second line.

The best way to do this would be:

    lines = ["Stats:"]
    for ...:
        line = ...
        lines.append(line)
    
    log.info("%s", os.linesep.join(lines))

File: rose/server/game.py [hunk 13]
```diff
@@ -96,9 +98,10 @@ def drive_player(self, name, info):
...
+        log.info(string)
```
Comment (nirs) [inline rose/server/game.py]: You should never use a format string including user input (player.name) - what if player.name includes a formatting placeholder? the logger will raise trying to format a missing argument. In the worst case, your log will be swallowed silently in runtime.

**Comment by nirs** [comment]
@moshe742, can you complete the PR?

**Comment by moshe742** [comment]
If all is good should I do the rebase?

**Comment by nirs** [comment]
@moshe742 I think the best way to complete this patch is:

1. squash all commits into one using:

   ```
    git rebase -i baba642abea9fe57c0f1b1cfd4e3d260de0f9315^
   ```
2. Update your master branch, should track upstream master

   ```
    git checkout master
    git pull upstream
   ```
   Assumes that you don't have any changes in the master branch - you should work on 
   your changes in a different branch.

   Assumes you have a remote called upstream, using this git repo:
   https://github.com/RedHat-Israel/ROSE.git

3. rebase on master:

   ```
    git rebase -i master
   ```

Finally push your changes with -f to update the PR.

Thanks!

**Comment by nirs** [comment]
@moshe742, I fixed several pep8 violations that your patch introduced, and fixed the stats order.

Merged it in 444e5941b83.
